xcube_gen_bc/iproc.py

```diff
@@ -112,3 +113,109 @@ class SnapOlciCyanoAlertL2InputProcessor(SnapNetcdfInputProcessor):
+                                xy_gcp_step=(max(1, len(dataset.lon) // 4),
+                                             max(1, len(dataset.lat) // 4)))
+
+    def get_time_range(self, dataset: xr.Dataset) -> Tuple[float, float]:
+        time_coverage_start, time_coverage_end = None, None
+        date_format = '%Y-%m-%d %H:%M:%S'
+        if "time" in dataset:
+            time_coverage_start = str(dataset.time[0].values)
+            date = pd.to_datetime(str(dataset.time[0].values), utc=True)
+            if 'daily' in str(dataset.attrs.values()):
+                time_coverage_end = datetime.datetime.strftime((date + datetime.timedelta(days=1)), date_format)
+            if 'hourly' in str(dataset.attrs.values()):
+                time_coverage_end = datetime.datetime.strftime((date + datetime.timedelta(hours=1)), date_format)
+
+        return to_time_in_days_since_1970(time_coverage_start), to_time_in_days_since_1970(time_coverage_end)
+
+    def _validate(self, dataset):
+        self._check_coordinate_var(dataset, "lon", min_length=2)
+        self._check_coordinate_var(dataset, "lat", min_length=2)
+        if "time" in dataset.dims:
+            self._check_coordinate_var(dataset, "time", max_length=1)
+            required_dims = ("time", "lat", "lon")
+        else:
+            required_dims = ("lat", "lon")
+        count = 0
+        for var_name in dataset.data_vars:
+            var = dataset.data_vars[var_name]
+            if var.dims == required_dims:
+                count += 1
+        if count == 0:
+            raise ValueError(f"dataset has no variables with required dimensions {required_dims!r}")
+
+    # noinspection PyMethodMayBeStatic
+    def _check_coordinate_var(self, dataset: xr.Dataset, coord_var_name: str,
+                              min_length: int = None, max_length: int = None):
+        if coord_var_name not in dataset.coords:
+            raise ValueError(f'missing coordinate variable "{coord_var_name}"')
+        coord_var = dataset.coords[coord_var_name]
+        if len(coord_var.shape) != 1:
+            raise ValueError('coordinate variable "lon" must be 1D')
+        coord_var_bnds_name = coord_var.attrs.get("bounds", coord_var_name + "_bnds")
+        if coord_var_bnds_name in dataset:
+            coord_bnds_var = dataset[coord_var_bnds_name]
+            expected_shape = (len(coord_var), 2)
+            if coord_bnds_var.shape != expected_shape:
+                raise ValueError(f'coordinate bounds variable "{coord_bnds_var}" must have shape {expected_shape!r}')
+        else:
+            if min_length is not None and len(coord_var) < min_length:
+                raise ValueError(f'coordinate variable "{coord_var_name}" must have at least {min_length} value(s)')
+            if max_length is not None and len(coord_var) > max_length:
+                raise ValueError(f'coordinate variable "{coord_var_name}" must have no more than {max_length} value(s)')
+
+
+def init_plugin():
```

Comment:
Remove this version of `init_plugin` here, the only valid one is in `plugin.py`.